tests/integration.rs

```diff
@@ -4356,6 +4356,97 @@ mod tests {
             handle_child_output(r, &output);
         }
 
+        fn get_balloon_mem(api_socket: &str) -> u64 {
+            let (cmd_success, cmd_output) = remote_command_w_output(&api_socket, "info", None);
+            assert!(cmd_success);
+
+            let info: serde_json::Value = serde_json::from_slice(&cmd_output).unwrap_or_default();
+            let total_mem = &info["config"]["memory"]["size"]
+                .to_string()
+                .parse::<u64>()
+                .unwrap();
+            let actual_mem = &info["memory_actual_size"]
+                .to_string()
+                .parse::<u64>()
+                .unwrap();
+            total_mem - actual_mem
+        }
+
+        #[test]
+        #[cfg(target_arch = "x86_64")]
+        fn test_virtio_balloon() {
+            let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+            let guest = Guest::new(Box::new(focal));
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let kernel_path = direct_kernel_boot_path();
+
+            let api_socket = temp_api_path(&guest.tmp_dir);
+
+            let guest_memory_size_gb = 32;
+
+            //Let's start a 32G guest with balloon occupied 10G memory
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=4"])
+                .args(&[
+                    "--memory",
+                    format!("size={}G", guest_memory_size_gb).as_str(),
+                ])
+                .args(&["--kernel", kernel_path.to_str().unwrap()])
+                .args(&["--cmdline", DIRECT_KERNEL_BOOT_CMDLINE])
+                .args(&["--balloon", "size=10737418240,deflate_on_oom=true"])
+                .default_disks()
+                .default_net()
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            let r = std::panic::catch_unwind(|| {
+                guest.wait_vm_boot(None).unwrap();
+
+                // Wait guest boots up, and then check balloon_mem
+                thread::sleep(std::time::Duration::new(10, 0));
```

Comment:
This should not be needed because of `wait_vm_boot()` right before.

Reply:
> This should not be needed because of `wait_vm_boot()` right before.
Actually, "--balloon size=10737418240" needs time to communicate with the guest after the guest starts up, or else orig_balloon do not have enough time to reach the desired 10737418240 bytes. Sorry that the "Wait guest boots up" comment do mislead. I will modify the comment later, and figure out how much time will be needed when the total size is 4G.

Reply:
> This should not be needed because of `wait_vm_boot()` right before.

Actually, "--balloon size=10737418240" needs time to communicate with the guest after the guest starts up, or else orig_balloon do not have enough time to reach the desired 10737418240 bytes. Sorry that the "Wait guest boots up" comment do mislead. I will modify the comment later, and figure out how much time will be needed when the total size is 4G.